tests/test-cases/Group1-Docker-Commands/1-38-Docker-Exec.robot

```diff
@@ -124,25 +124,128 @@ Exec NonExisting
+     Set Test Variable  ${ExecSimpleContainer}  Exec-simple-${suffix}
+     ${rc}  ${id}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run -itd --name ${ExecSimpleContainer} ${busybox} sleep 30
+     Should Be Equal As Integers  ${rc}  0
+
+     :FOR  ${idx}  IN RANGE  1  5
+     \   Start Process  docker %{VCH-PARAMS} exec ${id} /bin/ls  alias=exec-simple-%{VCH-NAME}-${idx}  shell=true
+
+     :FOR  ${idx}  IN RANGE  1  5
+     \   ${result}=  Wait For Process  exec-simple-%{VCH-NAME}-${idx}  timeout=40s
+     \   Should Be Equal As Integers  ${result.rc}  0
+     \   # if any of these are missing check to see if the busy box fs changed first.
+     \   Should Contain  ${result.stdout}  bin
+     \   Should Contain  ${result.stdout}  dev
+     \   Should Contain  ${result.stdout}  etc
+     \   Should Contain  ${result.stdout}  home
+     \   Should Contain  ${result.stdout}  lib
+     \   Should Contain  ${result.stdout}  lost+found
+     \   Should Contain  ${result.stdout}  mnt
+     \   Should Contain  ${result.stdout}  proc
+     \   Should Contain  ${result.stdout}  root
+     \   Should Contain  ${result.stdout}  run
+     \   Should Contain  ${result.stdout}  sbin
+     \   Should Contain  ${result.stdout}  sys
+     \   Should Contain  ${result.stdout}  tmp
+     \   Should Contain  ${result.stdout}  usr
+     \   Should Contain  ${result.stdout}  var
+
+     ${rc}=  Run And Return Rc  docker %{VCH-PARAMS} wait ${id}
+     Should Be Equal As Integers  ${rc}  0
+
+
+Exec During Poweroff Of A Container Performing A Long Running Task
+     ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} pull ${busybox}
+     Should Be Equal As Integers  ${rc}  0
+     Should Not Contain  ${output}  Error
+
+     ${suffix}=  Evaluate  '%{DRONE_BUILD_NUMBER}-' + str(random.randint(1000,9999))  modules=random
+     Set Test Variable  ${ExecPowerOffContainerLong}  Exec-Poweroff-${suffix}
+     ${rc}  ${id}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run -itd --name ${ExecPoweroffContainerLong} ${busybox} /bin/top
+     Should Be Equal As Integers  ${rc}  0
+
+     :FOR  ${idx}  IN RANGE  1  10
+     \   Start Process  docker %{VCH-PARAMS} exec ${id} /bin/ls  alias=exec-%{VCH-NAME}-${idx}  shell=true
+
+
+     Sleep  10s
+     ${rc}  ${id}=  Run And Return Rc And Output  docker %{VCH-PARAMS} stop ${id}
+     Should Be Equal As Integers  ${rc}  0
+
+     ${combinedErr}=  Set Variable
+     ${combinedOut}=  Set Variable
+
+     :FOR  ${idx}  IN RANGE  1  10
+     \   ${result}=  Wait For Process  exec-%{VCH-NAME}-${idx}  timeout=2 mins
+     \   ${combinedErr}=  Catenate  ${combinedErr}  ${result.stderr}${\n}
+     \   ${combinedOut}=  Catenate  ${combinedOut}  ${result.stdout}${\n}
+
+     Should Contain  ${combinedErr}  Container (${id}) is not running
+
+     # We should get atleast one successful exec...
+     Should Contain  ${combinedOut}  bin
```

Comment:
pull this out into a separate keyword - it's significant duplication in just these two tests.

Reply:
For this I would expect at least one success. So the keyword itself should probably check for both 1 and 2. in combination. `Simple Exec` is not meant to fail. If we abstract it out to a keyword then I can add the OR logic there in an effort to reduce the success checking in general. but having an OR on success and failure may mean that test never fails.

Reply:
agreed on the keyword breakout. agreed on the OR. by my reading of this test it is completely valid to get returned back 10 valid errors and never see the directory output - especially if the first exec takes more than 10s to happen..